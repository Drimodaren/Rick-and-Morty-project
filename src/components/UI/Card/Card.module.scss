.Card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 5px;
    background-color: #efefef;
    width: 240px;
    min-height: 128px;
    border-radius: 7px;
    box-shadow: 0px 5px 8px -3px var(--main-box-color);
    overflow: hidden;
    font-style: normal;
    font-family: "Roboto";
    text-decoration: none;
    h1 {
        font-size: 20px;
        font-weight: 500;
        line-height: 30px;
        letter-spacing: 0.15px;
        color: rgba(0, 0, 0, 0.87);
    }
    h2 {
        font-size: 14px;
        font-weight: 400;
        line-height: 21px;
        letter-spacing: 0.25px;
        color: rgba(0, 0, 0, 0.6);
    }
    h3 {
        font-size: 14px;
        font-weight: 700;
        line-height: 21px;
        letter-spacing: 0.25px;
        color: rgba(0, 0, 0, 0.6);
    }
    img {
        width: 240px;
    }
}
.CardImage {
    min-height: 244px;
    align-items: flex-start;
    h1 {
        padding-left: 16px;
        padding-top: 12px;
        gap: 1px;
    }
    h2 {
        padding-left: 16px;
        padding-bottom: 12px;
        gap: 1px;
    }
}
