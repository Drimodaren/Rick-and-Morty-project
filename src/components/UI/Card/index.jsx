import React, { useEffect } from "react";
import style from "./Card.module.scss";
import clsx from "clsx";
import { useDispatch, useSelector } from "react-redux";
import { loadCharaters } from "../../../store/characters/actions";

export default function Card({ url, title, description, info, image, id }) {
   
    return (
        <div className={clsx(style.Card, image && style.CardImage)} data-testid={`Card-${id}`}>
            {image && <img src={image} alt="Card" />}
            <h1>{title}</h1>
            <h2>{description}</h2>
            {info && <h3>{info}</h3>}
        </div>
    );
}
