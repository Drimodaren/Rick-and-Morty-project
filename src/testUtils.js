import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

const BrowserRouterProvider = ({ children }) => {
    return <BrowserRouter>{children}</BrowserRouter>;
};

export const withBrowserRouterRender = (ui, options) => render(ui, { wrapper: BrowserRouterProvider, ...options });
