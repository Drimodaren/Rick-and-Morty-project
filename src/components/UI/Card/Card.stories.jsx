import Card from "./index";


export default { title: "Card", component: Card };
const Template = args => <Card {...args} />;
export const Default = Template.bind({});
Default.args = { title: "Rick", description: "Human" };
export const CardWithOutImage = Template.bind({});
CardWithOutImage.args = { ...Default.args };
export const CardWithOutImageAndWithInfo = Template.bind({});
CardWithOutImageAndWithInfo.args = { ...Default.args, info: "Bastard" };
export const CardWithImage = Template.bind({});
CardWithImage.args = { ...Default.args, image: "https://rickandmortyapi.com/api/character/avatar/1.jpeg" };
