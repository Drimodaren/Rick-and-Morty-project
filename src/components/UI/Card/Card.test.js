import { screen } from "@testing-library/react";
import Card from ".";
import { withBrowserRouterRender } from "testUtils";

describe("cardComponent", () => {
    it("should render card withOut image", () => {
        withBrowserRouterRender(<Card url={""} title="Rick" description="Human" info="Bastard" id={2} />);
        expect(screen.getByTestId("Card-2")).toHaveTextContent("Rick");
        expect(screen.getByTestId("Card-2")).toHaveTextContent("Human");
        expect(screen.getByTestId("Card-2")).toHaveTextContent("Bastard");
        expect(screen.getByTestId("Card-2")).not.toHaveTextContent("111");
        expect(screen.getByTestId("Card-2")).toMatchSnapshot();
    });
    it("should render card with image", () => {
        withBrowserRouterRender(
            <Card
                url={""}
                title="Rick"
                description="Human"
                id={2}
                image="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
        );
        expect(screen.getByTestId("Card-2")).toHaveTextContent("Rick");
        expect(screen.getByTestId("Card-2")).toHaveTextContent("Human");
        expect(screen.getByRole("img")).toBeDefined();
        expect(screen.getByTestId("Card-2")).not.toHaveTextContent("Bastard");
        expect(screen.getByTestId("Card-2")).not.toHaveTextContent("111");
        expect(screen.getByTestId("Card-2")).toMatchSnapshot();
        //expect(screen.getByTestId("Card-2").querySelector('img')).toBeDefined();
    });
});
