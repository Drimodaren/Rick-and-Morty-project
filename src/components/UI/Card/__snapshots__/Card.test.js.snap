// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cardComponent should render card with image 1`] = `
<a
  class="Card CardImage"
  data-testid="Card-2"
  href="/2"
>
  <img
    alt="Card"
    src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
  />
  <h1>
    Rick
  </h1>
  <h2>
    Human
  </h2>
</a>
`;

exports[`cardComponent should render card withOut image 1`] = `
<a
  class="Card"
  data-testid="Card-2"
  href="/2"
>
  <h1>
    Rick
  </h1>
  <h2>
    Human
  </h2>
  <h3>
    Bastard
  </h3>
</a>
`;
